### PR TITLE
Deprecated `VCPKG_PREFER_SYSTEM_LIBS`

### PR DESCRIPTION
**Pull request by dg0yt**
Factored out of https://github.com/microsoft/vcpkg/pull/21783.

- #### What does your PR fix?
  Deprecates `VCPKG_PREFER_SYSTEM_LIBS`, suggesting overlay ports as a reliable alternative.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  not needed

### CODE REVIEW

**Comment by dg0yt** [comment]
No world rebuild (and it is not needed anyways).

**Comment by JackBoosY** [comment]
cc @Neumann-A

**Comment by JackBoosY** [comment]
Related code / doc : https://github.com/microsoft/vcpkg/search?q=VCPKG_PREFER_SYSTEM_LIBS

File: docs/users/buildsystems/cmake-integration.md [hunk 1]
```diff
@@ -213,6 +213,8 @@ This variable can be set to a list of additional command line parameters to pass
 
 ### `VCPKG_PREFER_SYSTEM_LIBS`
 
+**This feature has been deprecated. Use empty overlay ports instead.**
```
Comment (Thomas1664) [inline docs/users/buildsystems/cmake-integration.md:216]: I think it would be better to add a link to the overlay documentation here.

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**